### PR TITLE
docs/FORMS-1042 openapi docs for document templates

### PR DESCRIPTION
**Pull request by WalterMoar**
# Description

Adding the OpenAPI documentation for the new document templates routes. This documentation was skipped so that the routes could be merged and the frontend work started.

## Types of changes

Documentation (non-breaking change with enhancements to documentation)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have run the npm script lint on the frontend and backend
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)
- [x] I have approval from the product owner for the contribution in this pull request

### CODE REVIEW

File: app/src/docs/v1.api-spec.yaml [hunk 1]
```diff
@@ -2028,7 +2228,7 @@ paths:
           schema:
             type: string
             format: uuid
-            example: c6455376-382c-439d-a811-0381a012d696
+            example: aeb3b705-1de5-4f4e-a4e6-0716b7671034
```
Comment (WalterMoar) [inline app/src/docs/v1.api-spec.yaml:2231]: The docs were using random UUIDs and this often meant that the docs would say that a request using on ID would have a response with a different ID. Standardizing on using the same UUID for all sample form IDs, form submission IDs, etc.

File: app/src/docs/v1.api-spec.yaml [hunk 2]
```diff
@@ -4155,20 +4417,51 @@ components:
               description: >-
                 The original file type of the `content` document template.
               example: txt
+    TimeStampCreate:
```
Comment (WalterMoar) [inline app/src/docs/v1.api-spec.yaml:4420]: The example responses are all using TimeStampUserData, which include an updatedBy value and a different updatedAt timestamp. However, when creating a new object this is not correct - the updatedBy is not defined and the updatedAt is the same as the createdAt. Made this correct for the document template docs but will have to fix the others later.

File: app/src/docs/v1.api-spec.yaml [hunk 3]
```diff
@@ -4155,20 +4417,51 @@ components:
...
+          description: >
+            The UTC date and time when the object was last modified.
+          example: '2020-06-04T18:49:20.672Z'
     TimeStampUserData:
       type: object
       properties:
         createdBy:
           type: string
+          description: >
+            The username of the person who created the object.
```
Comment (WalterMoar) [inline app/src/docs/v1.api-spec.yaml:4449]: We should always be using descriptions. Once these are all set up we can configure the extension to enforce it.

**Comment by WalterMoar** [comment]
New docs deployed here: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-1320/api/v1/docs#tag/Document-Template

**Review by vijaivir — APPROVED** [review]
Looks good! I have also updated the API call so I'm not relying on the template from the /forms/{formId}/documentTemplates anymore.

**Pull request merged**